pkg/tarsum/tarsum_test.go

```diff
@@ -84,27 +84,27 @@ var testLayers = []testLayer{
 		tarsum:   "tarsum+sha256:cbe4dee79fe979d69c16c2bccd032e3205716a562f4a3c1ca1cbeed7b256eb19"},
 	{
 		options: &sizedOptions{1, 1024 * 1024, false, false}, // a 1mb file (in memory)
-		tarsum:  "tarsum+md5:0d7529ec7a8360155b48134b8e599f53",
+		tarsum:  "tarsum+md5:3a6cdb475d90459ac0d3280703d17be2",
 		hash:    md5THash,
 	},
 	{
 		options: &sizedOptions{1, 1024 * 1024, false, false}, // a 1mb file (in memory)
-		tarsum:  "tarsum+sha1:f1fee39c5925807ff75ef1925e7a23be444ba4df",
+		tarsum:  "tarsum+sha1:14b5e0d12a0c50a4281e86e92153fa06d55d00c6",
 		hash:    sha1Hash,
 	},
 	{
 		options: &sizedOptions{1, 1024 * 1024, false, false}, // a 1mb file (in memory)
-		tarsum:  "tarsum+sha224:6319390c0b061d639085d8748b14cd55f697cf9313805218b21cf61c",
+		tarsum:  "tarsum+sha224:dd8925b7a4c71b13f3a68a0f9428a757c76b93752c398f272a9062d5",
 		hash:    sha224Hash,
 	},
 	{
 		options: &sizedOptions{1, 1024 * 1024, false, false}, // a 1mb file (in memory)
-		tarsum:  "tarsum+sha384:a578ce3ce29a2ae03b8ed7c26f47d0f75b4fc849557c62454be4b5ffd66ba021e713b48ce71e947b43aab57afd5a7636",
+		tarsum:  "tarsum+sha384:e39e82f40005134bed13fb632d1a5f2aa4675c9ddb4a136fbcec202797e68d2f635e1200dee2e3a8d7f69d54d3f2fd27",
```

Comment:
ping @tonistiigi @dmcgowan for input/impact

Reply:
It seems that `FileInfoHeader()` from `pkg/archive` is not affected, as it uses `tar.FileInfoHeader()` which explicitly sets `Typeflag = TypeReg` for regular files.

Reply:
In other words, tarballs generated by docker always have `Typeflag = TypeReg` for regular files (quickly checked with `docker save`).

Reply:
> for input/impact

Reply:
I think the goal previous times was to not break the build cache too many times; given that we'll break the cache when we make the move to buildkit (non experimental), likely we want to avoid breaking the cache before that, but indeed could use Tõnis input on that

Reply:
From my understanding (see analisys in comments above), build cache won't be broken -- the checksum issue is limited to this particular test. But this is theory; is there any way to quickly check it?